torch/csrc/jit/codegen/cuda/executor.cpp

```diff
@@ -246,15 +250,41 @@ at::Tensor inferAndAlloc(
   // NOLINTNEXTLINE(cppcoreguidelines-init-variables)
   std::vector<int64_t> inferred_sizes;
 
-  for (const auto size : sizes) {
-    const auto inferred_val = expr_eval.evaluate(size);
-    TORCH_INTERNAL_ASSERT(
-        inferred_val.has_value(),
-        "Could not launch kernel as program could not infer ",
-        kir::toString(size),
-        " for the buffer ",
-        kir::toString(tv));
-    inferred_sizes.push_back(inferred_val.value());
+  if (opt_memoized_sizes.has_value()) {
```

Comment:
Similarly, I guess this part of logic could go inside of the evaluator itself. We wouldn't want `inferAndAlloc` to take on too much work on optimized integer evaluation.